pinot-server/src/main/java/org/apache/pinot/server/api/resources/TablesResource.java

```diff
@@ -781,6 +781,7 @@ public String uploadLLCSegment(
       String realtimeTableName,
       @ApiParam(value = "Name of the segment", required = true) @PathParam("segmentName") String segmentName,
       @QueryParam("uploadTimeoutMs") @DefaultValue("-1") int timeoutMs,
+      @QueryParam("expectedCrc") @DefaultValue("-1") long expectedCrc,
```

Comment:
I think it might be better to add a new API that returns the CRC for a given segment on the server.

That API could be used for a bunch of other things and it will also make the deepstore upload retry task cleaner

Reply:
Yes makes sense to me! Updated the patch accordingly!